File: src/test/scala/com/martomate/tripaint/model/coords/StorageCoordsTest.scala
package com.martomate.tripaint.model.coords

import munit.FunSuite

class StorageCoordsTest extends FunSuite {
  test("constructor should require x >= 0") {
    intercept[IllegalArgumentException](StorageCoords(-1, 0))
    intercept[IllegalArgumentException](StorageCoords(-10, 0))
    intercept[IllegalArgumentException](StorageCoords(-10, 100))
    StorageCoords(0, 10)
  }
  test("constructor should require y >= 0") {
    intercept[IllegalArgumentException](StorageCoords(0, -1))
    intercept[IllegalArgumentException](StorageCoords(0, -10))
    intercept[IllegalArgumentException](StorageCoords(100, -10))
    StorageCoords(10, 0)
  }
}

File: core/src/test/kotlin/tripaint/coords/StorageCoordsTest.kt
package tripaint.coords

import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.assertThrows
import kotlin.test.Test

class StorageCoordsTest {
    @Nested
    inner class Construction {
        @Test
        fun `requires x is at least 0`() {
            assertThrows<IllegalArgumentException> { StorageCoords.from(-1, 0) }
            assertThrows<IllegalArgumentException> { StorageCoords.from(-10, 0) }
            assertThrows<IllegalArgumentException> { StorageCoords.from(-10, 100) }
            StorageCoords.from(0, 10)
        }

        @Test
        fun `requires y is at least 0`() {
            assertThrows<IllegalArgumentException> { StorageCoords.from(0, -1) }
            assertThrows<IllegalArgumentException> { StorageCoords.from(0, -10) }
            assertThrows<IllegalArgumentException> { StorageCoords.from(100, -10) }
            StorageCoords.from(10, 0)
        }
    }
}
